loadtest/control/simulcontroller/actions.go

```diff
@@ -1282,3 +1300,357 @@ func sendTypingEventIfEnabled(u user.User, channelId string) error {
 	}
 	return nil
 }
+
+func (c *SimulController) viewGlobalThreads(u user.User) control.UserActionResponse {
+	collapsedThreads, resp := control.CollapsedThreadsEnabled(u)
+	if resp.Err != nil || !collapsedThreads {
+		return resp
+	}
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("viewGlobalThreads: current team should be set"))}
+	}
+
+	// View "All your threads" in the Global Threads Screen
+	threads, err := u.Store().ThreadsSorted(false, false)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if len(threads) == 0 {
+		threads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     false,
+			Since:      0,
+			TotalsOnly: false,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	oldestThreadId := threads[len(threads)-1].PostId
+	// scrolling between 1 and 3 times
+	numScrolls := rand.Intn(3) + 1
+	for i := 0; i < numScrolls; i++ {
+		threads, err = u.GetUserThreads(team.Id, &model.GetUserThreadsOpts{
+			PageSize:   25,
+			Extended:   false,
+			Deleted:    false,
+			Unread:     false,
+			Since:      0,
+			TotalsOnly: false,
+			Before:     oldestThreadId,
+		})
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		oldestThreadId = threads[len(threads)-1].PostId
+		// idle time between scrolls, between 1 and 10 seconds.
+		idleTime := time.Duration(1+rand.Intn(10)) * time.Second
+		select {
+		case <-c.stopChan:
+			return control.UserActionResponse{Info: "action canceled"}
+		case <-time.After(idleTime):
+		}
+	}
+
+	// Switch to "Unread" tabs
+	unreadThreads, err := u.Store().ThreadsSorted(true, false)
```

Comment:
The API end-point is the same, the parameter `unread` option is the only thing that changes. I couldn't figure out how to filter the metric by parameter in grafana to correctly weight the API calls. 
Since the "unreads" tab is only accessible through the Global Threads Screen, I added both calls in the same action.